packages/core/src/components/pressable.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { ComponentProps } from 'react'
+import { styled } from '../css/styled'
+import { Pressable as NativePressable, Platform } from 'react-native'
+
+declare module 'react-native' {
+  interface PressableStateCallbackType {
+    hovered?: boolean
+    focused?: boolean
+  }
+}
+
+const StyledPressable = styled(NativePressable)({})
+const Press = React.forwardRef(function Pressable(
+  { sx = {}, disabled, ...props }: ComponentProps<typeof StyledPressable>,
+  ref: ComponentProps<typeof NativePressable>['ref']
+) {
+  return (
+    <StyledPressable
+      // TODO: Pressable accepts a function as a style with the computed properties figure out a way to pass these along
```

Comment:
I wonder if we just omit the style prop and give a warning.

Reply:
Do you think it would somehow be possible to detect a function is getting passed and pass that along in the createThemeComponent or css functions? 

I'm not quite sure how it would work yet. Just a thought.

Reply:
That would be nice, but unfortunately I don't see how. Our css function already allows the style/sx prop to be a function, which receives the theme. The only way to achieve this would be to somehow call the style function inline, which isn't possible. 

The only thing we could technically do is make our own Pressable by tracking onPressIn/onPressOut, as well as onHoverIn/onHoverOut.

But at that point, it's a ton of re-renders for what I view as little benefit. I know it kinda sucks to lose the style as a function prop, but unless we put a View as the child, I don't see a way around it. And doing that isn't great, because suddenly you're styling something that's wrapped in another element.

Reply:
I'll just omit the style prop for the meantime and maybe a better solution will come up later down the line.

Reply:
Let's omit style from the TS types too, using `Omit<Props, 'style'>`.

In the next major release, we should probably ignore the `style` prop and not apply `css` to it. Instead, it should get passed as-is to the wrapped component.

In hindsight, it was a mistake to not do this from the start. Luckily the docs only show usage with `sx`, so maybe that's what people expect to happen. But we need to leave `style` to be a plain prop. This would give us the opportunity to build a Babel plugin that takes the `sx` prop and turns it into `StyleSheet.create` and passes the styles to `style`, for instance.

Reply:
100% agree. I think I may have asked this in the past, would a custom babel plugin give us the ability to apply media queries to the CSS styles directly, replacing fresnel, or am I missing something? 

I know absolutely nothing about babel I'll attempt to help if possible.

Reply:
Yeah I'm totally new to Babel too, but I've been slowly getting acquainted with how it works.

The gist is, you can manipulate code at the place it is written. So for example, you could build a plug-in that reads every `sx` prop in a file, removes the sx prop, turns into StyleSheet.create at the bottom of the file, and then passes it to the `style` prop. In order to apply the theme to it, you would have to pass the theme object as a config variable of the Babel plugin. This is because it all occurs at build-time. You statically look at a file, and say, okay, do this to this code.

A nice helper is called Babel macros. They're inline Babel plugins that appear to be much simpler and have an easier API. You import a macro as a function and then call it. Whereas a normal Babel plugin is "magical" -- you don't import it in your code, it just works in the background.

So, a long answer: a Babel plugin with the current Fresnel setup could not support responsive arrays, since all it would do is turn things into a stylesheet before they get passed to a Dripsy component. So basically, if I were building the plugin, I'd check if there are any arrays, and if there are, I would skip over that prop. It's possible we could figure that out if we got rid of SSR support via Fresnel. 

The big irony of my Fresnel endeavor is that my Next.js app doesn't even use SSR in the end, since one of my providers at the root of the website returns null before it's mounted. I haven't noted the site to be any slower because of this. And Google still crawls websites that use JS to render now. So that's something to keep in mind for the future - is Fresnel the needed solution or not for web. I guess it depends how many people actually use it to render on the server.

Reply:
<img width="1907" alt="Screen Shot 2021-04-18 at 5 59 19 PM" src="https://user-images.githubusercontent.com/13172299/115162152-eba1de00-a06f-11eb-8a45-8ec7e72970ca.png">

Here's an example of what Babel's input looks like.

https://astexplorer.net/#/gist/2a0b2d1d36eef922630fa188b463994f/14c7be9883cb1dc15b787dc32a9bd6ad02041527

Reply:
Thanks for the explanation. That makes a ton of sense. 

It looks like snackui would support SSR via babel transforming there `useMedia` hook to CSS, but correct me if I'm wrong (You may already know this). 

I guess that's a pretty valid point. I'm still curious as to what extent google cares about an SSR page vs a client rendered page. I'd be a little worried even though they do scrape javascript rendered pages that in some scenarios it might still decrease the overall SEO value of the page. But then again who knows.

Overall have you run into any Fresnel-related problems with it rendering multiple divs? I haven't had anything super substantial but it can be pretty annoying when your styles don't seem to work sometime and takes some getting used to.

Reply:
Oh interesting, is snack compiling down to actual media queries? I thought they were using JS queries but I can ask Nate who maintains it.

I've basically stopped using the Fresnel altogether. I still have it in my legacy code, but I now use a Sizer component that has inline style props, and on web renders a `div`. 

It's basically a `SkinnyView`. I'll add this and the same for Text to a future release.

Reply:
> Oh interesting, is snack compiling down to actual media queries? I thought they were using JS queries but I can ask Nate who maintains it.

I actually thought the same thing until I checked the docs recently and saw that it looks like they are compiling to CSS now which I thought was super cool.

Reply:
I see, I don't see that anywhere with media queries, maybe I missed it. I do see that the rest compiles down to CSS

Reply:
Ahhh I think you're right. 

I was looking at the roadmap here: https://github.com/snackui/snackui/blob/master/ROADMAP.md#advanced-traversal. And saw his future implementation of the "scaling" feature.

Reply:
Makes sense. We can move this discussion into an easier to find issue. For now I'll just try memoizing styles and see how that works.

Reply:
Before I forget: maybe we could add a `elementType?: 'text' | 'view'` option in `createThemedComponent`.

If an element is a `view` type, and a given component only receives an `sx` prop and `children`, then on web it could render the `SkinnyView`, removing the need for fresnel in those cases. Maybe not a great idea because if you add a prop on the fly, it would change to fresnel which would then alter the behavior between renders. But I bet there's a creative solution like that to optimize it. Maybe there could be a warning if it toggles between them.

Or, perhaps just a user relying on `SkinnyView` is better.

Reply:
On that note, the latest version of Dripsy has big time perf improvements. All styles are now memoized and use class names instead of like styles 

Vía https://github.com/nandorojo/dripsy/issues/88